### ISSUE TITLE
What hash functions are currently supported?

### ISSUE THREAD

**Issue by camper42** [housepower/spark-clickhouse-connector#114]
We encountered `xenon.clickhouse.exception.ClickHouseClientException: [-1] Unsupported ClickHouse expression: FuncExpr[cityHash64(XXXX)]` while writing to a table with a `sample by` expression.

same exception with `murmurHash2_64`

What hash functions are currently supported? https://clickhouse.com/docs/en/sql-reference/functions/hash-functions/

create table stmt:
```
CREATE TABLE dev.sample_test
(
    `did` String,
    `load_date` Date
)
ENGINE = MergeTree
PRIMARY KEY xxHash64(did)
PARTITION BY load_date
SAMPLE BY xxHash64(did)
SETTINGS index_granularity = 8192
```

**Comment by pan3793** [comment]
I think the problem should cause by `PRIMARY KEY xxHash64(did)` because the connector will try to repartition by shard keys and partition keys, and sort by primary keys before writing.
Currently, Spark 3.2/3.3 itself does not support complex `Transform` other than `IdentityTransform` on writing. Since SPARK-37377(available since 3.3), Spark can recognize the V2Function on reading.

**Comment by pan3793** [comment]
There is an idea comes from #89 to expose a configuration to make the repartition optional, it's also suitable for sort.
I think we can introduce 3 configurations for those cases,
`spark.clickhouse.write.repartitionByShardingKeys`
`spark.clickhouse.write.repartitionByPartitions`
`spark.clickhouse.write.localSortByPrimaryKeys`

**Comment by pan3793** [comment]
`spark.clickhouse.write.localSortByKey` has been add in master branch, will be shipped in 0.3.0 (after Spark 3.3.0 GA)

**Comment by pan3793** [comment]
Fixed by #116 #119

**Comment by camper42** [comment]
Thanks for your work😃
